src/inmanta/agent/executor.py

```diff
@@ -175,7 +175,7 @@ def blueprint_hash(self) -> str:
                 "pip_config": self.pip_config.dict(),
                 "requirements": self.requirements,
                 # Use the hash values of the sources, sorted to ensure consistent ordering
-                "sources": sorted(list(set(source.hash_value for source in self.sources))),
+                "sources": sorted(source.hash_value for source in self.sources),
```

Comment:
Actually, I think we don't even want to sort, because again we would have cases where we get the same hash for different executors (e.g. file `a.py` renamed to `z.py` -> `sources` order is different from `source.hash_value` order).

As an extension to this, file rename `a0.py` to `a1.py` would still suffer from this behavior, because we do not include module/file names in the blueprint hash at all. But that's not new in this PR.

Reply:
But that means hashing the file name into it as well?